pkg/model/components/kubedns.go

```diff
@@ -74,5 +74,33 @@ func (b *KubeDnsOptionsBuilder) BuildOptions(o interface{}) error {
 		clusterSpec.KubeDNS.MemoryLimit = &defaultMemoryLimit
 	}
 
+	NodeLocalDNS := clusterSpec.KubeDNS.NodeLocalDNS
+	if NodeLocalDNS == nil {
```

Comment:
What is the benefit of doing this vs just keep checking if NodeLocalDNS is nil?

Reply:
You mean to assign this assignment?

```golang
NodeLocalDNS := clusterSpec.KubeDNS.NodeLocalDNS
```

Reply:
Sorry, I mean these two:
```
NodeLocalDNS = &kops.NodeLocalDNSConfig{}
NodeLocalDNS.Enabled = false
```
In general there is a lot of manipulation of the spec itself here, which would unconditionally override what the users may set. I would avoid all of these.

Reply:
But my understanding is that this logic runs only when a user does not set anything in `NodeLocalDNS` spec so I don't get how the override should happen. Could you elaborate please?

Reply:
fyi I'm following the pattern here https://github.com/mazzy89/kops/blob/nodelocaldnscache/pkg/model/components/kubedns.go#L37

Reply:
I guess that is fine when the values are empty. Although it doesn't look very common to do it this way anymore. Could be that it is done this way for kube-dns either because the addon is very old or because this addon is not optional.

Either way, I am fine with this approach as long as user values are not overridden. This does make the templates more simple.